src/LayerPlan.cpp

```diff
@@ -1227,6 +1227,109 @@ void LayerPlan::addSplitWall(
+
+        const double coasting_volume = extruder_settings.get<double>("coasting_volume");
+        const double coasting_min_volume = extruder_settings.get<double>("coasting_min_volume");
+
+        for (const auto& reversed_chunk : paths | ranges::views::enumerate | ranges::views::reverse
+                                              | ranges::views::chunk_by(
+                                                  [](const auto&path_a, const auto&path_b)
+                                                  {
+                                                      return (! std::get<1>(path_a).isTravelPath()) || std::get<1>(path_b).isTravelPath();
+                                                  }))
+        {
+            double accumulated_volume = 0.0;
+            bool chunk_coasting_point_reached = false;
+            bool chunk_min_volume_reached = false;
+
+            for (const auto& [path_idx, path] : reversed_chunk)
+            {
+                if (path.isTravelPath())
+                {
+                    continue;
+                }
+
+                PathCoasting& path_coasting = path_coastings[path_idx];
+                const double path_extrusion_per_length = path.config.extrusion_mm3_per_mm * path.flow;
+
+                for (const auto& [point_idx, point] : path.points | ranges::views::enumerate | ranges::views::reverse)
+                {
+                    Point3LL previous_point;
+                    if (point_idx > 0)
+                    {
+                        previous_point = path.points[point_idx - 1];
+                    }
+                    else if (path_idx > 0)
+                    {
+                        previous_point = paths[path_idx - 1].points.back();
+                    }
+                    else
+                    {
+                        previous_point = current_position;
+                    }
+
+                    const double segment_length = INT2MM((point - previous_point).vSize());
+                    const double segment_volume = segment_length * path_extrusion_per_length;
+
+                    accumulated_volume += segment_volume;
+
+                    if (! chunk_coasting_point_reached && path_coasting.apply_coasting == ApplyCoasting::NoCoasting)
+                    {
+                        if (accumulated_volume >= coasting_volume)
+                        {
+                            const double start_pos_factor = (accumulated_volume - coasting_volume) / segment_volume;
+                            Point3LL coasting_start_pos = cura::lerp(previous_point, point, start_pos_factor);
+                            path_coasting = { ApplyCoasting::PartialCoasting, point_idx, coasting_start_pos };
+                            chunk_coasting_point_reached = true;
+                        }
+                        else if (point_idx == 0)
```

Comment:
This could do with a comment -- it's not really clear that we're looking at the end here because it's easy to miss that we reversed the path.